src/main/java/org/jalasoft/moi/model/core/ICommandBuilder.java

```diff
@@ -25,4 +26,10 @@
      * @return String
      */
     String buildCommand(Path path);
+
+    String buildCommand(List<Path> paths);
+
+    String commandToRun(Path path);
+
+    String buildCommandFolder(Path path);
```

Comment:
I saw the `JavaCommandBuilder, CppCommandBuilder` and `CSharpCommandBuilder `not needed those methods yet.
If you change the interface, You'll force to implement to others class, I believe that is a good approach but we can discuss about with those who will be affected.

Reply:
If the other languages not need this method yet, the functionality should not be affected,

Reply:
I agree with you, but this changes the project's architecture, and other classes return `null`. I can propose some alternatives

- Create a new interface that extends from the previous one, and only the classes that use those behaviors implement the new interface.
- Implement each method in other classes so they can handle those behaviors

and there are many other ways you can choose.